src/Symfony/Component/CssSelector/Node/MatchingNode.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\CssSelector\Node;
+
+/**
+ * Represents a "<selector>:is(<subSelectorList>)" node.
+ *
+ * This component is a port of the Python cssselect library,
+ * which is copyright Ian Bicking, @see https://github.com/SimonSapin/cssselect.
+ *
+ * @author Hubert Lenoir <lenoir.hubert@gmail.com>
+ *
+ * @internal
+ */
+class MatchingNode extends AbstractNode
+{
+    private NodeInterface $selector;
+    /** @var array<NodeInterface> */
+    private array $arguments;
+
+    public function __construct(NodeInterface $selector, array $arguments = [])
```

Comment:
I think we can use constructor property promotion here, same for the other class.

Reply:
You're right. It's better like this!